src/VisualStudio/Core/Def/Implementation/Remote/RemoteHostClientServiceFactory.RemoteHostClientService.cs

```diff
@@ -213,26 +207,6 @@ private void OnConnectionChanged(object sender, bool connected)
                 FatalError.Report(new Exception("Connection to remote host closed"));
             }
 
-            private bool FeaturesEnabled()
-            {
-                if (ServiceFeatureOnOffOptions.IsClosedFileDiagnosticsEnabled(_workspace.Options, LanguageNames.CSharp))
```

Comment:
I didn't verify, but i believe this code would have caused us to load the C# roslyn bits even if you were only opening VB or Typescript.